aten/src/ATen/native/cuda/IndexKernel.cu

```diff
@@ -234,7 +234,6 @@ static void index_copy_kernel(
 
 
 static void index_put_kernel(TensorIterator& iter, const IntArrayRef index_size, const IntArrayRef index_stride, const bool accumulate) {
-  TORCH_CHECK(!accumulate, "index_put does not support accumulate=true");
```

Comment:
but how come was this check not hit before?

Reply:
Exactly, it looks shady. Looking more into it.

Reply:
I get this! Interesting.

So when the device type is cuda and accumulate=True, the operator never calls `index_put_kernel`; it always calls `index_put_with_sort_kernel`. Here:

https://github.com/pytorch/pytorch/blob/ff026f3d0a68a4533b193498adc5b5c7b045b73c/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L814-L819

And `index_put_with_sort_kernel` does not contain any such checks. Here:
https://github.com/pytorch/pytorch/blob/ff026f3d0a68a4533b193498adc5b5c7b045b73c/aten/src/ATen/native/cuda/Indexing.cu#L429

I think, therefore, we were never able to catch this earlier. What do you think? Thanks!

Reply:
Fair, but in that case we should do an `TORCH_INTERNAL_ASSERT` but keep the check.

Reply:
Sure, I've updated the PR.